recipes/nccl/build.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+################################################
+# Use `asm-generic/socket.h` from distro.      #
+# Needed to ensure `SO_REUSEPORT` is defined.  #
+# This feature was added in Linux kernel 3.9.  #
+# However RHEL 6 and CentOS 6 backported it.   #
+# The sysroot compilers get in the way so we   #
+# use this header from this system to define   #
+# `SO_REUSEPORT`.                              #
+#                                              #
+# ref: https://lwn.net/Articles/542629/        #
+################################################
+CONDA_BUILD_SYSROOT="$(${CC} --print-sysroot)"
+cp /usr/include/asm-generic/socket.h "${CONDA_BUILD_SYSROOT}/usr/include/asm-generic/socket.h"
+
+make -j${CPU_COUNT} CUDA_HOME="${CUDA_HOME}" CUDARTLIB="cudart"
+make install PREFIX="${PREFIX}"
+
+# Delete the static library as it is quite large.
+# Should halve the package size.
+rm "${PREFIX}/lib/libnccl_static.a"
```

Comment:
Split output? I imagine there could be packages that want to static link nccl, but this could be a future enhancement.

Reply:
Good point. Given the Anaconda packages already ship with this, we should just drop this line short-term for compatibility. Will open an issue on the feedstock for a proper split package.

Reply:
Have readded the static library in the latest commit.

Reply:
Raised as issue ( https://github.com/conda-forge/nccl-feedstock/issues/6 ).